x-pack/solutions/security/plugins/security_solution/public/management/components/console_argument_selectors/custom_scripts_selector/custom_script_selector.tsx

```diff
@@ -58,229 +43,198 @@ export interface CustomScriptSelectorState<TScriptRecordMeta extends {} = {}> {
-          <EuiToolTip position="right" content={option.description}>
-            <EuiText data-test-subj={`${option.label}-description`} color="subdued" size="s">
-              <small css={truncationStyle}>{option.description}</small>
-            </EuiText>
-          </EuiToolTip>
-        )}
-      </div>
-    );
-  };
+  const state = useBaseSelectorState(store, value);
+  const { handleOpenPopover, handleClosePopover } = useBaseSelectorHandlers(
+    state,
+    onChange,
+    value || '',
+    valueText || ''
+  );
 
-  const handleOpenPopover = useCallback(() => {
-    setIsPopoverOpen(true);
-  }, [setIsPopoverOpen]);
+  const isAwaitingRenderDelay = useRenderDelay();
 
-  const handleClosePopover = useCallback(() => {
-    setIsPopoverOpen(false);
-  }, [setIsPopoverOpen]);
+  useFocusManagement(state.isPopoverOpen, requestFocus);
 
-  // Focus on the console's input element when the popover closes
-  useEffect(() => {
-    if (!state.isPopoverOpen && requestFocus) {
-      // Use setTimeout to ensure focus happens after the popover closes
-      setTimeout(() => {
-        requestFocus();
-      }, 0);
-    }
-  }, [state.isPopoverOpen, requestFocus]);
+  useGenericErrorToast(error, notifications, ERROR_LOADING_CUSTOM_SCRIPTS);
 
-  const handleScriptSelection = useCallback(
+  const handleSelection = useCallback(
     (newOptions: EuiSelectableOption[], _event: unknown, changedOption: EuiSelectableOption) => {
-      if (changedOption.checked === 'on') {
-        onChange({
-          value: changedOption.label,
-          valueText: changedOption.label,
-          store: {
-            ...state,
-            isPopoverOpen: false,
-            selectedOption: changedOption.data as ResponseActionScript,
-          },
-        });
-      } else {
-        onChange({
-          value: '',
-          valueText: '',
-          store: {
-            ...state,
-            isPopoverOpen: false,
-            selectedOption: undefined,
-          },
-        });
-      }
+      const handler = createSelectionHandler(onChange, state, handleSentinelOneSelectionChange);
+      handler(newOptions, _event, changedOption);
     },
-    [onChange, state]
+    [onChange, state, handleSentinelOneSelectionChange]
   );
 
-  // notifications comes from useKibana() and is of type NotificationsStart
-  // which is compatible with our updated useCustomScriptsErrorToast function
-  useCustomScriptsErrorToast(scriptsError, notifications);
+  const renderOption = useCallback(
+    (option: EuiSelectableOption) => {
+      const hasDescription = 'description' in option && option.description;
+      const hasToolTipContent = 'toolTipContent' in option && option.toolTipContent;
+      const testIdPrefix = testId();
+      const descriptionText = hasDescription ? String(option.description) : '';
+      const toolTipText = hasToolTipContent ? String(option.toolTipContent) : '';
+
+      const content = (
+        <div data-test-subj={`${testIdPrefix}-script`}>
```

Comment:
(optional) you can just call `testId('script')` here and everywhere else instead of doing string templates